lib/jxl/dec_frame.h

```diff
@@ -89,6 +89,29 @@ class FrameDecoder {
     kPartial = 3,
   };
 
+  // Currently only the following combinations of progressive detail are
+  // implemented:
+  // - kDC | kFrames
+  // - kPasses | kLastPasses | kDC | kFrames
+  enum ProgressiveDetail {
+    // never return JXL_DEC_FRAME_PROGRESSION
+    kNone = 0,
+    // after completed kRegularFrames
+    kFrames = 1,
+    // after completed DC (1:8)
+    kDC = 2,
+    // after completed AC passes that are the last pass for their resolution
+    // target.
+    kLastPasses = 4,
+    // after completed AC passes that are not the last pass for their resolution
+    // target.
+    kPasses = 8,
+    // during DC Frame when lower resolution are completed (1:32, 1:16)
+    kDCProgressive = 16,
+    // after completed groups
+    kGroups = 32,
+  };
```

Comment:
So maybe to make it somewhat easier to use, kLastPasses could imply kDC (because DC can be seen as the 'zero-th' pass for the 1:8 resolution target), kPasses could imply kLastPasses, and kGroups could imply everything else.

Reply:
I'm making it linear scale, where higher values imply smaller ones.